presidio-analyzer/presidio_analyzer/predefined_recognizers/uk_nino_recognizer.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class UkNinoRecognizer(PatternRecognizer):
+    """
+    Recognizes UK National Insurance Number using regex.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS = [
+        Pattern(
+            "NINO (medium)",
```

Comment:
Couldc you please provide the source of this regex? How did you test it aside from the added unit tests?

Reply:
The UK Government defines the National Insurance Number format [Here](https://www.gov.uk/hmrc-internal-manuals/national-insurance-manual/nim39110). I wrote this taking some inspiration from t[his Stackoverflow discussion](https://stackoverflow.com/questions/10204378/regular-expression-to-validate-uk-national-insurance-number/17779536).

As the National Insurance Number is case insensitive and Presido allows users to set regex flags at global/registry levels, I included both upper and lower case characters in the regex.